mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/MessageMapperTest.java

```diff
@@ -377,6 +378,23 @@ void deleteShouldNotUpdateRecentWhenNotNeeded() throws MailboxException {
         assertThat(messageMapper.findRecentMessageUidsInMailbox(benwaInboxMailbox)).containsOnly(message2.getUid(), message4.getUid());
     }
 
+    @Test
+    void deleteMessagesShouldCorrectlyUpdateMailboxCounters() throws MailboxException {
+        saveMessages();
+        messageMapper.updateFlags(benwaInboxMailbox, new FlagsUpdateCalculator(new Flags(Flag.SEEN), FlagsUpdateMode.REPLACE), MessageRange.one(message2.getUid()));
```

Comment:
could we write a decent api for changing a flag?

Reply:
That's orthogonal to the bug reolution but let's add a default method then

Reply:
I wrote a :

```
    default Optional<UpdatedFlags> updateFlags(Mailbox mailbox, MessageUid uid, FlagsUpdateCalculator flagsUpdateCalculator) throws MailboxException {
        return Iterators.toStream(updateFlags(mailbox, flagsUpdateCalculator, MessageRange.one(uid)))
            .findFirst();
    }
```

Reply:
I was thinking about:
```
messageMapper.updateFlags(benwaInboxMailbox, message2.getUid(), flags -> flags.set(Flag.SEEN));
```

Reply:
That's not an easy change to unify `flags -> flags.set(Flag.SEEN)` into `FlagsUpdateCalculator`

Don't you mind if I don't invest for now in that refactoring?

Reply:
no problem

Reply:
look at https://github.com/linagora/james-project/pull/3537 and pick it if you want

Reply:
I will but in a follow up PR